src/js/libcs/List.js

```diff
@@ -209,9 +209,9 @@ List.consecutive = function(a) {
 };
 
 List.reverse = function(a) {
-    var erg = [];
-    for (var i = a.value.length - 1; i >= 0; i--) {
-        erg.push(a.value[i]);
+    var erg = new Array(a.value.length);
```

Comment:
I did some tests beforehand:
```

var n = 25000000;
var a = new Array(n);
for(let i = 0; i < n; i++) a[i] = i;

var t = Date.now();
var erg = [];
for (let i = a.length-1 ; i >= 0; i--) {
        erg.push(a[i]);
    }
console.log("time push", Date.now() -t);

t = Date.now();
var erg = new Array(a.length);
for (let i = a.length-1, j = 0; i >= 0; i--, j++) {
        erg[j]= a[i];
    }
console.log("time native", Date.now() -t);

t = Date.now();
erg = a.slice().reverse();
console.log("time slice reverse", Date.now()-t);
```
which resulted in:
```
time push 414
time native 169
time slice reverse 406
```
on Chromium. Would be great to have some iOS device tested.